addons/website/static/src/components/dialog/edit_menu.js

```diff
@@ -59,21 +131,51 @@ export class MenuDialog extends Component {
                 },
                 urlChosen: () => {
                     this.url.input.value = input.value;
+                    this.state.pageNotFound = false;
                 },
             };
             const unmountAutocompleteWithPages = wUtils.autocompleteWithPages(input, options, this.env);
             return () => unmountAutocompleteWithPages();
         }, () => [this.urlInputRef.el]);
+
+        onMounted(() => {
+            if (!this.props.isMegaMenu) {
+                this.state.pageNotFound =
+                    isPageNotFound(this.urlInputRef.el.value, this.allPages);
+            }
+            this.state.urlPlaceholder = wUtils.slugify(this.name.input.value);
+        });
     }
 
     onClickOk() {
         if (this.name.isValid()) {
             if (this.props.isMegaMenu || this.url.isValid()) {
```

Comment:
You can avoid non necessary nesting here, and returns early.
```suggestion
            const nameValid = this.name.isValid();
            const urlValid = this.props.isMegaMenu || this.url.isValid();

            if (!nameValid || !urlValid) return;
```